sockeye/decoder.py

```diff
@@ -158,7 +162,15 @@ def __init__(self,
             self.mapped_context_b = mx.sym.Variable("%smapped_context_bias" % prefix)
 
         # Stacked RNN
-        self.rnn = rnn.get_stacked_rnn(self.rnn_config, self.prefix)
+        if self.rnn_config.num_layers == 1 or not self.attention_in_upper_layers:
+            self.attention_rnn = rnn.get_stacked_rnn(self.rnn_config, self.prefix, False)
```

Comment:
maybe its easiest to read if "rnn" is the prefix of these vairables. Something like rnn_layer_1 ?
rnn_attention vs. rnn_upper?